### ISSUE TITLE
Multiday Appointment

### ISSUE THREAD

**Issue by KevinKrueger** [danheron/Heron.MudCalendar#37]
It should be recognizable that an appointment can go over days instead of drawing it each time as a single appointment.

![image](https://github.com/danheron/Heron.MudCalendar/assets/25382167/d2d23b99-55c7-456e-89c1-39978bdc4efa)

**Comment by KevinKrueger** [comment]
Hey Dan, how's it looking?
I know you're doing this on the side. 
I'm familiar with it.

But that would be really important for me.

**Comment by danheron** [comment]
I'm quite busy at the moment so I can't give you an ETA on that.  However it's an open source project so you can fork it and add any extra functionality that you need.

**Comment by KevinKrueger** [comment]
What we need:
- [ ] Drag-Areas for resizing start/end
- [ ] Mouse-Over handle
- [ ] evtl. DragStates (None, Start, End, Both) 
- [ ] Mouse pointer changes when these areas are entered

Should help with implementation (No matter who does it in the end):
![image](https://github.com/danheron/Heron.MudCalendar/assets/25382167/bcf89cd7-2d8b-490a-ae47-b951ae0328ad)

**Comment by Int32Overflow** [comment]
@KevinKrueger Is there already a plan for implementation? I could also help.

**Comment by pcrosthwaite** [comment]
I'd also vote for this. Not sure I could help though

**Comment by KevinKrueger** [comment]
> @KevinKrueger Is there already a plan for implementation? I could also help.

Well, it should work in the same way as in the Outlook calendar.

**Comment by Int32Overflow** [comment]
I've tried a bit, but unfortunately haven't found a good solution yet :-(

For this experiment, I tried overlays with the help of the grid. But then MudDragDrop no longer works...
![grafik](https://github.com/danheron/Heron.MudCalendar/assets/20139038/bae4a717-97f6-4525-baf8-f05139ec6413)
Here the example code:
![grafik](https://github.com/danheron/Heron.MudCalendar/assets/20139038/5f63950c-b5e8-49c4-81ef-e3b7adf38210)

@danheron  Before I go any further and look for another solution, we need to resolve the limitations:
### Limitations
- Scrolling (overflow-x) of a cell: This is logically not possible. Another solution would be a manual overflow:
![grafik](https://github.com/danheron/Heron.MudCalendar/assets/20139038/9056d44a-5e3a-43ad-9479-23af607c0ba0)

**Comment by danheron** [comment]
I agree that removing scrolling would be better.  I have created another issue to make this change: #119

If we are going to make the Month View more complicated, such as adding multiday appointments, then I think an overall redesign is needed.  The Day/Week view is built with 2 layers:  The first layer draws the cells, then the second layer draws the appointments on top of the cells.  This way the appointments are not constrained to a single cell.  I think we should do something similar with the Month view.

**Comment by arunnagar1991** [comment]
Please guide us how to achieve above functionality, or any progress with the above?

**Comment by danheron** [comment]
@arunnagar1991 It's like I wrote - 2 layers like the Day/Week view.  Look at the code for the Day/Week view to see how to do it.

I can do it, but I won't have time for a few weeks.

**Comment by marcusjm** [comment]
Is this still considered for 2.0?

**Comment by dennisrahmen** [comment]
Hey just FYI I am using this calendar: https://github.com/valincius/BlazorScheduler
It has the multi day functionality you want so maybe you can use it as inspiration.

**Comment by marcusjm** [comment]
> Hey just FYI I am using this calendar: https://github.com/valincius/BlazorScheduler It has the multi day functionality you want so maybe you can use it as inspiration.

Is that project still maintained? Might be an option.

**Comment by dennisrahmen** [comment]
@marcusjm I think he has moved on to other projects.

**Comment by marcusjm** [comment]
> @marcusjm I think he has moved on to other projects.

Yes, looks no better. I am simply using FullCalendar for now but I hope this Project gets some of the additions, it is much nicer with BlazorFirst approach :).

**Comment by danleash** [comment]
> > @marcusjm I think he has moved on to other projects.
> 
> Yes, looks no better. I am simply using FullCalendar for now but I hope this Project gets some of the additions, it is much nicer with BlazorFirst approach :).
> 
> Lets face it. Heron Calendar would be perfect it added this "last" thing. I don't need anything else. Let's hope @danheron gets around to it.

You realize you are on Github and the project is open source right? Why don't _you_ "get around to it" lol.

**Comment by dennisrahmen** [comment]
Also I meant like inspiration to use some code snippets in this project, to get the multiday functionality.

Since, as I said, he moved on to other projects.

**Comment by marcusjm** [comment]
> > > @marcusjm I think he has moved on to other projects.
> > 
> > 
> > Yes, looks no better. I am simply using FullCalendar for now but I hope this Project gets some of the additions, it is much nicer with BlazorFirst approach :).
> > Lets face it. Heron Calendar would be perfect it added this "last" thing. I don't need anything else. Let's hope @danheron gets around to it.
> 
> You realize you are on Github and the project is open source right? Why don't _you_ "get around to it" lol.

If it was that simple. That requires a) That you have very good knowledge of every aspects of this component b) Also use a coding style where a PR would be readily accepted. 

That is how you realise you are on GitHub ;).

**Comment by KevinKrueger** [comment]
One good and one bad news story. The good news is that I was really pleased with the email I received and the bad news is that it doesn't work in your examples.(I have taken the current development status and compiled it)

Multiday Tests:
![image](https://github.com/user-attachments/assets/1445c324-f7e7-4d30-8834-cd8a3b29a46b)

![image](https://github.com/user-attachments/assets/72519f1f-ba35-4cc6-b2a3-2a4cef49ac49)
Resize  (Error):
Uncaught TypeError: Cannot read properties of undefined (reading 'clientHeight')
    at HTMLDivElement.onMouseDown (Heron.MudCalendar.min.js:1:3586)
onMouseDown @ Heron.MudCalendar.min.js:1Understand this errorAI

**Comment by danheron** [comment]
Sorry, I keep forgetting I need to make it work with the German culture where a comma is used as a decimal point.

I've fixed it now with PR #190

**Comment by danheron** [comment]
1. The ability to stretch an appointment across multiple weeks is not a trivial change and won't make it into the next release.

2. I cannot recreate this problem.  Can you tell me:
Which browser are you using?
Under the calendar there is some text that says "Item Changed....".  Is the end time updating correctly?

**Comment by danheron** [comment]
I think Visual Studio isn't stopping for a breakpoint - it's stopping because there is an error.  After the error the ResizeFinished event handler isn't being called and the end date of the item isn't being updated.

But the error doesn't seem to be in MudCalendar, it looks like something to do with BrowserLink.

Do you also get this problem when you run without debugging?

**Comment by KevinKrueger** [comment]
The problem persists even when running without debug mode.

**Comment by danheron** [comment]
I'll reopen the issue, but so far I've been unable to recreate the problem.

**Comment by danheron** [comment]
Closing this because I can't recreate it and nobody else has reported the same problem.
